Dockerfile

```diff
@@ -1,9 +1,17 @@
 FROM python:3.8
 
-WORKDIR /app
+RUN useradd -ms /bin/bash app
 
-COPY requirements.txt requirements.txt
-RUN pip3 install -r requirements.txt
+USER app
 
-COPY *.py /app/
-COPY custom_types.json /app/
+WORKDIR /home/app
+
+ENV PATH="/home/app/.local/bin:${PATH}"
+
+COPY --chown=app:app requirements.txt /home/app/requirements.txt
+COPY --chown=app:app *.py /home/app/
```

Comment:
Move this line to the bottom of the page (after install requirements). It will speed up local development, requirements.txt really changes, so you can reuse it as Docker cache.

Reply:
sure